### PR TITLE
Adaptation for working with Exonum version 0.9

### PR DESCRIPTION
**Pull request by aleksuss**
## Overview

Please describe your changes here and list any open questions you might have.

---
See: https://jira.bf.local/browse/XYZ

### Definition of Done

- [ ] There are no TODOs left in the code
- [ ] Change is covered by automated [tests](https://github.com/exonum/exonum-java-binding/blob/master/CONTRIBUTING.md#tests)
- [ ] The [coding guidelines](https://github.com/exonum/exonum-java-binding/blob/master/CONTRIBUTING.md#the-code-style) are followed
- [ ] Public API has Javadoc
- [ ] Method preconditions are checked and documented in the Javadoc of the method
- [ ] The [continuous integration build](https://www.travis-ci.org/exonum/exonum-java-binding) passes

### CODE REVIEW

File: exonum-java-binding-core/rust/src/proxy/service.rs [hunk 1]
```diff
@@ -1,15 +1,16 @@
-use exonum::blockchain::{ApiContext, Service, Transaction};
+use exonum::api::{self, ServiceApiBuilder};
```
Comment (DarkEld3r) [inline exonum-java-binding-core/rust/src/proxy/service.rs]:
```diff
-  use exonum::api::{self, ServiceApiBuilder}; 
+  use exonum::api::{self, ServiceApiBuilder, ServiceApiState};
```
?

File: exonum-java-binding-core/rust/src/runtime/java_service_runtime.rs [hunk 2]
```diff
@@ -117,12 +117,15 @@ pub fn panic_if_java_options() {
 pub struct JavaServiceFactory;
 
 impl ServiceFactory for JavaServiceFactory {
-    fn command(&mut self, command: &str) -> Option<Box<CommandExtension>> {
-        use exonum::helpers::fabric;
+    fn service_name(&self) -> &str {
+        "JAVA_SERVICE_FACTORY"
```
Comment (DarkEld3r) [inline exonum-java-binding-core/rust/src/runtime/java_service_runtime.rs]: I'm not sure this would be so simple... :upside_down_face:

Comment (aleksuss) [reply exonum-java-binding-core/rust/src/runtime/java_service_runtime.rs]: It's a temporary workaround and doesn't related to the main problem 😄

File: exonum-java-binding-core/rust/src/proxy/service.rs (hunk 1 above)
Comment (aleksuss) [reply exonum-java-binding-core/rust/src/proxy/service.rs]: What's confusing you ? Yes, we begun to use `ServiceApiState` in closure at 159 line.

**Comment by vitvakatu** [comment]
I thought it would be much more difficult :D 
Looks good to me, but I think it's impossible to merge right now (because we need to update our compiler version first). And we also need to fix this issue with factory name. I think EJB team will handle the rest of it, if you don't mind, @aleksuss. Thank you!

**Comment by aleksuss** [comment]
@vitvakatu you are welcome. I suppose this PR can help you to create a temporary workaround for bumping version of Exonum up to 0.9.

File: exonum-java-binding-core/rust/src/proxy/service.rs [hunk 3]
```diff
@@ -151,26 +151,32 @@ impl Service for ServiceProxy {
...
-    fn public_api_handler(&self, context: &ApiContext) -> Option<Box<Handler>> {
-        let node = NodeContext::new(
-            self.exec.clone(),
-            context.blockchain().clone(),
-            *context.public_key(),
-            context.node_channel().clone(),
+    fn wire_api(&self, builder: &mut ServiceApiBuilder) {
+        let self_ = self.clone();
+        builder.public_scope().endpoint(
+            "",
```
Comment (dmitry-timofeev) [inline exonum-java-binding-core/rust/src/proxy/service.rs]: Would you please clarify what "" is? Does this method have any unintended side-effects (e.g., creates a new handler that processes (?) each request)?

**CC** @alekseysidorov

File: exonum-java-binding-core/rust/src/runtime/java_service_runtime.rs (hunk 2 above)
Comment (dmitry-timofeev) [reply exonum-java-binding-core/rust/src/runtime/java_service_runtime.rs]: Is it supposed to be the name of the service (it seems sensible given this method name), or could it be distinct from it? If the first, why are there two such methods?

**Comment by aleksuss** [comment]
This PR can help https://github.com/exonum/exonum/pull/929 
It adds `blockchain` which contains `api_sender` into ServiceApiBuilder

**Comment by dmitry-timofeev** [comment]
> This PR can help exonum/exonum#929
It adds blockchain which contains api_sender into ServiceApiBuilder

Thanks! @vitvakatu is already working on updating the EJB to 0.9.x :tada:

**Pull request closed without merging**